src/pydata_sphinx_theme/__init__.py

```diff
@@ -92,22 +95,34 @@ def update_config(app, env):
 
         # try to read the json file. If it's a url we use request,
         # else we simply read the local file from the source directory
-        # it will raise an error if the file does not exist
+        # display a log warning if the file cannot be reached
+        content = None
         if urlparse(json_url).scheme in ["http", "https"]:
-            content = requests.get(json_url).text
+            try:
+                request = requests.get(json_url)
+                request.raise_for_status()
+                content = request.text
+            except (ConnectionError, HTTPError, RetryError):
+                pass
         else:
-            content = Path(env.srcdir, json_url).read_text()
-
-        # check that the json file is not illformed
-        # it will throw an error if there is a an issue
-        switcher_content = json.loads(content)
-        missing_url = any(["url" not in e for e in switcher_content])
-        missing_version = any(["version" not in e for e in switcher_content])
-        if missing_url or missing_version:
-            raise AttributeError(
-                f'The version switcher "{json_url}" file is malformed'
-                ' at least one of the items is missing the "url" or "version" key'
-            )
+            try:
+                content = Path(env.srcdir, json_url).read_text()
+            except FileNotFoundError:
+                pass
+
+        if content is None:
+            logger.warning(f'The version switcher "{json_url}" file cannot be read.')
```

Comment:
here I'd be inclined to capture the reason for the failure (in the `except` clauses above) and expand the logger message to say *why* it cannot be read. For example

> The version switcher foo.json file could not be read due to ThisKindOfError